aten/src/ATen/native/quantized/cpu/qnnpack/src/convolution.c

```diff
@@ -475,11 +475,16 @@ enum pytorch_qnnp_status pytorch_qnnp_create_convolution2d_nhwc_q8(
         pytorch_qnnp_compute_requantization_params(
             convolution_scale, output_zero_point, output_min, output_max);
   } else {
+    // Passing address of kernel_zero_point and convolution_scale
+    // is not safe as they are stack variables. However for now just staging this change.
+    // Eventually, we will find a better way to handle this.
+    // For now we dont go through this path for convolutions.
+    // TODO Kimish
```

Comment:
If I understand this code correctly I think this usage is perfectly fine since you are dealing with a blocking function.  Unless you hold onto these pointers internally after this function returns and use them subsequently, which is indeed unsafe behavior, there is no way the variables these pointers point to can go out of scope before the function exits.

Reply:
Yes so usage is fine. I just didnt feel comfortable passing an address of the stack variable. But you are right. It is safe otherwise.

Reply:
Let me remove this comment.